### PR TITLE
fix: prefer disabling imports with `imports.autoImport`

### PR DESCRIPTION
**Pull request by wattanx**
### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation or readme)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description
The `#imports` alias is set inside the import module.
So even if `imports: false`, I think you need to run import module.

In Nuxt 3, I believe the import module is executed even with autoImport: false.
https://github.com/nuxt/nuxt/blob/master/packages/nuxt/src/core/nuxt.ts#L406-L412

(I am wondering if I should be able to set it with the `imports.autoImport` option like in Nuxt 3.)

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/bridge/src/imports/module.ts [hunk 1]
```diff
@@ -114,7 +115,7 @@ export default defineNuxtModule<Partial<ImportsOptions>>({
   }
 })
 
-function addDeclarationTemplates (ctx: Unimport) {
+function addDeclarationTemplates (ctx: Unimport, options: ImportsOptions) {
```
Comment (danielroe) [inline packages/bridge/src/imports/module.ts:118]: this doesn't seem to be used

```suggestion
function addDeclarationTemplates (ctx: Unimport) {
```

File: packages/bridge/src/imports/module.ts [hunk 2]
```diff
@@ -91,7 +92,7 @@ export default defineNuxtModule<Partial<ImportsOptions>>({
     await regenerateImports()
 
     // Generate types
-    addDeclarationTemplates(ctx)
+    addDeclarationTemplates(ctx, options)
```
Comment (danielroe) [inline packages/bridge/src/imports/module.ts:95]:
```suggestion
    addDeclarationTemplates(ctx)
```

Comment (wattanx) [reply packages/bridge/src/imports/module.ts:95]: The `options` are used here!

https://github.com/nuxt/bridge/blob/dfd7beee88e306ceef165a24500d71fe801a9e9b/packages/bridge/src/imports/module.ts#L144-L151

Fixed to be the same as Nuxt 3. Is it unnecessary?
https://github.com/nuxt/nuxt/blob/main/packages/nuxt/src/imports/module.ts#L161-L168

Comment (danielroe) [reply packages/bridge/src/imports/module.ts:95]: you're quite right!

**Review by danielroe — CHANGES_REQUESTED** [review]
I think we should probably still respect `imports: false` but warn when this is enabled, and suggest instead that they set `imports.autoImports: false`.

**Comment by wattanx** [comment]
@danielroe 
Thanks for the review!
I agree with your opinion!
I added a warning in the following commit.
https://github.com/nuxt/bridge/pull/853/commits/1dfe5629691b68c405fb289d6b7fd9f86414b040

File: packages/bridge/src/module.ts [hunk 3]
```diff
@@ -68,26 +67,39 @@ export default defineNuxtModule({
...
     }
+
+    // @ts-expect-error TODO: legacy module container usage
+    nuxt.hook('modules:done', moduleContainer =>
+      installModule(
+        importsModule.bind(moduleContainer, {
+          autoImport: nuxt.options.imports?.autoImport ?? opts.imports ?? opts.autoImports
+        })
+      )
+    )
```
Comment (danielroe) [inline packages/bridge/src/module.ts:101]:
```suggestion
	  if (opts.imports ?? opts.autoImports) {
	    // @ts-expect-error TODO: legacy module container usage
	    nuxt.hook('modules:done', moduleContainer =>
	      installModule(
	        importsModule.bind(moduleContainer, {
	          autoImport: nuxt.options.imports?.autoImport ?? opts.imports ?? opts.autoImports
	        })
	      )
	    )
	  }
```

File: packages/bridge/src/module.ts [hunk 4]
```diff
@@ -25,7 +25,6 @@ export default defineNuxtModule({
     app: {},
     capi: {},
     transpile: true,
-    imports: true,
     compatibility: true,
```
Comment (danielroe) [inline packages/bridge/src/module.ts:28]:
```suggestion
    imports: true,
    compatibility: true,
```

**Review by danielroe — APPROVED** [review]

**Pull request merged**